test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -1243,7 +1242,7 @@ bool UcxConnection::send_common(const void *buffer, size_t length,
 
     assert(_ucx_status == UCS_OK);
 
-    ucp_request_param_t param;
+    ucp_request_param_t &param = _context.param;
```

Comment:
we should make a global param var only for params that could be initialized statically (once) and not changed later. and make a different param var for each case.
This way we can pre-initialize param struct and then just use it during send/recv.
It's not possible when some param fields depend on send function parameters - for example, when passing memh.
Also, it's not needed for slow-path case such as stream operation for exchanging conn_id.

Reply:
@yosefe Could we define class type and constructor to initialize ucp_request_param_t in c++ program? I find it must initialize all the structure members one by one explicitly under ```__cplusplus 199711L```. For example, it needs to initialize all the members until ```.cb.recv``` if we only want to initialize ```op_attr_mask``` and ```cb.recv``` explicitly.

Reply:
@changchengx can make these structs static member variables and initialize them like global variables.

Reply:
1. For slow path(such as ```connect_tag``` which uses ```stream recv/send```), keep the original code there
2. For the ```ucp_request_param_t type object``` that depends on function parameter(such as ```memh```), keep the original code there.
3. Define ```static ucp_request_param_t UcxContext::param``` object and initialize the object in ```UcxContext::init``` function. Then, use this object in ```UcxContext::recv_io_message``` function without constructing it again.

Reply:
@yosefe Passed all CI test cases.